docs/guides/private-submodules.rst

```diff
@@ -0,0 +1,92 @@
+Using Private Git Submodules with Read the Docs for Business
```

Comment:
Not sure about adding `with Read the Docs for Business` in the title, just in case someone tries to use it for .org? IDK :man_shrugging:

Reply:
It makes sense to me to mention this somewhere in this page. Maybe if the title is too long (and also will be duplicated in the Index as well) we could a warning/note immediately after the title and before the first paragraph.